geniza/corpus/solr_queryset.py

```diff
@@ -219,6 +226,19 @@ def keyword_search(self, search_term):
             )
         return search
 
+    def regex_search(self, search_term):
+        """Build a Lucene query for searching with regex."""
+        # store original regex query
+        original_regex = search_term
+        # surround passed query with wildcards to allow non-anchored matches,
+        # and slashes so that it is interpreted as regex by Lucene
+        search_term = f"/.*{search_term}.*/"
+        # match in the non-analyzed transcription_regex field
+        search = self.search(f"transcription_regex:{search_term}").raw_query_parameters(
+            regex_query=original_regex
```

Comment:
My thinking was that it's needed in `get_highlighting` so we can pass the unmodified regex query to the post-search python highlighting function (so that it matches only the relevant portion as opposed to the whole result as required by Lucene). Storing it instead as a property on the `DocumentSolrQuerySet` instance doesn't seem to work, as it's always set back to its original value in `get_highlighting`—perhaps because `get_highlighting` is called on the paginated result instead of the original queryset? It's still able to get any raw params we passed to solr, so that was my quick and dirty solution.

Maybe a better solution, since we don't do any additional modifications to the regex query, would just be to check for `transcription_regex:/.*` and strip that and the trailing `.*/` from the query to retrieve the original regex query.

Reply:
If there's a way to get both versions of the value you need from one place, that seems better.

If you need to persist information across filtered querysets, you can extend the `_clone` method that's used for the fluid interface: https://github.com/Princeton-CDH/parasolr/blob/main/parasolr/query/queryset.py#L612